packages/block-editor/src/components/use-moving-animation/index.js

```diff
@@ -42,114 +39,113 @@ const getAbsolutePosition = ( element ) => {
+		isAncestorMultiSelected,
+	} = useSelect( blockEditorStore );
+
+	// Whenever the trigger changes, we need to take a snapshot of the current
+	// position of the block to use it as a destination point for the animation.
+	const { previous, prevRect } = useMemo(
+		() => ( {
+			previous: ref.current && getAbsolutePosition( ref.current ),
+			prevRect: ref.current && ref.current.getBoundingClientRect(),
+		} ),
+		// eslint-disable-next-line react-hooks/exhaustive-deps
 		[ triggerAnimationOnChange ]
 	);
 
-	// Calculate the previous position of the block relative to the viewport and
-	// return a function to maintain that position by scrolling.
-	const preserveScrollPosition = useMemo( () => {
-		if ( ! adjustScrolling || ! ref.current ) {
-			return () => {};
+	useLayoutEffect( () => {
+		if ( ! previous || ! ref.current ) {
+			return;
 		}
 
 		const scrollContainer = getScrollContainer( ref.current );
-
-		if ( ! scrollContainer ) {
-			return () => {};
-		}
-
-		const prevRect = ref.current.getBoundingClientRect();
-		return () => {
-			const blockRect = ref.current.getBoundingClientRect();
-			const diff = blockRect.top - prevRect.top;
-
-			if ( diff ) {
-				scrollContainer.scrollTop += diff;
+		const isSelected = isBlockSelected( clientId );
+		const adjustScrolling =
+			isSelected || isFirstMultiSelectedBlock( clientId );
+
+		function preserveScrollPosition() {
+			if ( adjustScrolling && prevRect ) {
+				const blockRect = ref.current.getBoundingClientRect();
+				const diff = blockRect.top - prevRect.top;
+
+				if ( diff ) {
+					scrollContainer.scrollTop += diff;
+				}
 			}
-		};
-	}, [ triggerAnimationOnChange, adjustScrolling ] );
-
-	useLayoutEffect( () => {
-		if ( triggeredAnimation ) {
-			endAnimation();
-		}
-	}, [ triggeredAnimation ] );
-	useLayoutEffect( () => {
-		if ( ! previous ) {
-			return;
 		}
 
-		if ( prefersReducedMotion ) {
+		if (
+			window.matchMedia( '(prefers-reduced-motion: reduce)' ).matches ||
+			isTyping() ||
+			getGlobalBlockCount() > BLOCK_ANIMATION_THRESHOLD
+		) {
 			// If the animation is disabled and the scroll needs to be adjusted,
 			// just move directly to the final scroll position.
 			preserveScrollPosition();
-
 			return;
 		}
 
+		const isPartOfSelection =
+			isSelected ||
+			isBlockMultiSelected( clientId ) ||
+			isAncestorMultiSelected( clientId );
+		const zIndex = isPartOfSelection ? '1' : '';
+
+		const controller = new Controller( {
+			x: 0,
+			y: 0,
+			config: { mass: 5, tension: 2000, friction: 200 },
+			onChange( { value } ) {
+				if ( ! ref.current ) {
+					return;
+				}
+				let { x, y } = value;
+				x = Math.round( x );
+				y = Math.round( y );
+				const finishedMoving = x === 0 && y === 0;
+				ref.current.style.transformOrigin = 'center center';
+				ref.current.style.transform = finishedMoving
+					? null // Set to `null` to explicitly remove the transform.
+					: `translate3d(${ x }px,${ y }px,0)`;
+				ref.current.style.zIndex = zIndex;
```

Comment:
Maybe we can also set this at before animation start, then restore at the end or on cleanup.

Reply:
(Same for transformOrigin)

Reply:
I left the original code here. Didn't want to alter things too much and I was hoping to see a before start and stop API.